javalib/src/main/scala/java/lang/Math.scala

```diff
@@ -1,324 +1,317 @@
+    cmath.remainder(f1, f2)
+
+  @inline def incrementExact(a: scala.Int): scala.Int =
+    addExact(a, 1)
+
+  @inline def incrementExact(a: scala.Long): scala.Long =
+    addExact(a, 1L)
+
+  @inline def log(a: scala.Double): scala.Double =
+    `llvm.log.f64`(a)
+
+  @inline def log10(a: scala.Double): scala.Double =
+    `llvm.log10.f64`(a)
+
+  @inline def log1p(a: scala.Double): scala.Double =
+    cmath.log1p(a)
+
+  @inline def max(a: scala.Double, b: scala.Double): scala.Double =
+    `llvm.maxnum.f64`(a, b)
+
+  @inline def max(a: scala.Float, b: scala.Float): scala.Float =
+    `llvm.maxnum.f32`(a, b)
+
+  @inline def max(a: scala.Int, b: scala.Int): scala.Int =
+    if (a > b) a else b
+
+  @inline def max(a: scala.Long, b: scala.Long): scala.Long =
+    if (a > b) a else b
+
+  @inline def min(a: scala.Double, b: scala.Double): scala.Double =
+    `llvm.maxnum.f64`(a, b)
+
+  @inline def min(a: scala.Float, b: scala.Float): scala.Float =
+    `llvm.minnum.f32`(a, b)
+
+  @inline def min(a: scala.Int, b: scala.Int): scala.Int =
+    if (a < b) a else b
+
+  @inline def min(a: scala.Long, b: scala.Long): scala.Long =
+    if (a < b) a else b
+
+  @inline def multiplyExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val overflow = `llvm.smul.with.overflow.i32`(a, b)
+    if (overflow.flag) throw new ArithmeticException("Integer overflow")
+    else overflow.value
   }
 
-  def subtractExact(a: scala.Long, b: scala.Long): scala.Long = {
-    val res       = a - b
-    val resSgnBit = res < 0
-    if (resSgnBit == (a < 0) || resSgnBit == (b > 0)) res
-    else throw new ArithmeticException("Long overflow")
+  @inline def multiplyExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val overflow = `llvm.smul.with.overflow.i64`(a, b)
+    if (overflow.flag) throw new ArithmeticException("Long overflow")
+    else overflow.value
   }
 
-  def multiplyExact(a: scala.Int, b: scala.Int): scala.Int = {
-    val overflow = {
-      if (b > 0)
-        a > Integer.MAX_VALUE / b || a < Integer.MIN_VALUE / b
-      else if (b < -1)
-        a > Integer.MIN_VALUE / b || a < Integer.MAX_VALUE / b
-      else if (b == -1)
-        a == Integer.MIN_VALUE
-      else
-        false
+  @inline def negateExact(a: scala.Int): scala.Int =
+    subtractExact(0, a)
+
+  @inline def negateExact(a: scala.Long): scala.Long =
+    subtractExact(0, a)
+
+  def nextAfter(a: scala.Float, b: scala.Double): scala.Float = {
+    val aabs = abs(a.toDouble)
+    val babs = abs(b)
+
+    if (Float.isNaN(a) || Double.isNaN(b)) {
+      Float.NaN
+    } else if (aabs == 0f && babs == 0d) {
+      b.toFloat
+    } else if (aabs == Float.MIN_VALUE && babs < aabs) {
+      copySign(0, a)
+    } else if (Float.isInfinite(a) && babs < aabs) {
+      copySign(Float.MAX_VALUE, a)
+    } else if (aabs == Float.MAX_VALUE && babs > aabs) {
+      copySign(Float.POSITIVE_INFINITY, a)
+    } else {
+      cmath.nextafterf(a, b.toFloat)
```

Comment:
Hum. Now that I see it written like that, I think there's also a similar problem, but reversed. Let's say I compute `nextAfter(1.5f, nextUp(1.5))`. In that case, `b > a`, so the result should be `nextUp(1.5f)`. However, `b.toFloat` rounds back down to `1.5f`, and therefore `cmath.nextafterf(a, b.toFloat)` will return `1.5f` instead.

Experiment on the JVM:
```scala
scala> Math.nextAfter(1.5f, Math.nextUp(1.5))
res0: Float = 1.5000001

scala> Math.nextAfter(1.5f, Math.nextUp(1.5).toFloat)
res1: Float = 1.5
```
So I guess you should explicitly perform
```scala
if (b > a) cmath.nextafterf(a, Float.POSITIVE_INFINITY)
else cmath.nextafterf(a, Float.NEGATIVE_INFINITY)
```
or something like that :-s